### ISSUE TITLE
Youtube shared links do not show up as messages

### ISSUE THREAD

**Issue by db-bogdan** [dequis/purple-facebook#354]
Whenever someone directly shares a Youtube video from a mobile device to my Facebook account, it does not inform me in any way that it has been received.

Log sent by mail.

**Comment by db-bogdan** [comment]
By the way, correct me if I'm wrong, @dequis but I believe this is the line of interest:

```
(22:07:13) facebook: Parsing JSON: {"deltas":[{"deltaNewMessage":{"messageMetadata":{"threadKey":{"otherUserFbId":100002307291089},"messageId":"mid.$cAAAAANK2S0Zjy1DLzldmgx1g9MmD","offlineThreadingId":6297865291128162691,"actorFbId":100002307291089,"timestamp":1501528032206,"shouldBuzzDevice":1,"tags":["source:titan:orca"]},"attachments":[{"id":"1445163965570499","fbid":1445163965570499,"xmaGraphQL":"{\"ZXh0ZW5zaWJsZV9tZXNzYWdlX2F0dGFjaG1lbnQ6MTQ0NTE2Mzk2NTU3MDQ5OQ\":{\"__type__\":{\"name\":\"ExtensibleMessageAttachment\"},\"id\":\"ZXh0ZW5zaWJsZV9tZXNzYWdlX2F0dGFjaG1lbnQ6MTQ0NTE2Mzk2NTU3MDQ5OQ\",\"is_forwardable\":true,\"story_attachment\":{\"deduplication_key\":\"ca5f3ea1140edc40d2813149791fb57a\",\"description\":{\"text\":\"David Gilmour's \\\"Rattle That Lock\\\" album released on September 18, 2015 Buy the album at http:\\/\\/smarturl.it\\/RattleThatLockAmzDLX Watch 'In Any Tongue' here: ...\"},\"media\":{\"__type__\":{\"name\":\"GenericAttachmentMedia\"},\"image\":{\"uri\":\"https:\\/\\/external.fotp3-1.fna.fbcdn.net\\/safe_image.php?d=AQAN1tyMUSrl4tOj&w=960&h=960&url=https\\u00253A\\u00252F\\u00252Fi.ytimg.com\\u00252Fvi\\u00252F7PwQrEbEnrM\\u00252Fmaxresdefault.jpg&cfs=1&_nc_hash=AQDuWpckE6Kl26DM\"}},\"properties\":[{\"attachment_property_type\":\"double\",\"key\":\"width\",\"value\":{\"text\":\"1920\"}},{\"attachment_property_type\":\"double\",\"key\":\"height\",\"value\":{\"text\":\"1080\"}}],\"source\":{\"text\":\"youtube.com\"},\"style_list\":[\"video_share_youtube\",\"video_share\",\"share\",\"fallback\"],\"target\":{\"__type__\":{\"name\":\"ExternalUrl\"}},\"title\":\"David Gilmour - The Girl In The Yellow Dress (Official Music Video)\",\"tracking\":\"{}\",\"url\":\"fbrpc:\\/\\/facebook\\/nativethirdparty?app_id=87741124305&app_name=YouTube&class_name=com.google.android.youtube.WatchActivity&fallback_url0=https\\u00253A\\u00252F\\u00252Fm.facebook.com\\u00252Fapps\\u00252Fyt-fb-app&market_uri=market\\u00253A\\u00252F\\u00252Fdetails\\u00253Fid\\u00253Dcom.google.android.youtube\\u002526referrer\\u00253Dutm_source\\u0025253Dapps.facebook.com\\u00252526utm_campaign\\u0025253Dfb4a\\u00252526utm_content\\u0025253D\\u002525257B\\u0025252522app\\u0025252522\\u002525253A87741124305\\u002525252C\\u0025252522t\\u0025252522\\u002525253A1501528032\\u002525257D&package_name=com.google.android.youtube&tap_behavior=app_fallback_web&target_url=https\\u00253A\\u00252F\\u00252Fwww.youtube.com\\u00252Fwatch\\u00253Fv\\u00253D7PwQrEbEnrM&extra_applink_key=al_applink_data&referer_data_key=extras&al_applink_data=\\u00257B\\u002522target_url\\u002522\\u00253A\\u002522https\\u00253A\\u00255C\\u00252F\\u00255C\\u00252Fwww.youtube.com\\u00255C\\u00252Fwatch\\u00253Fv\\u00253D7PwQrEbEnrM\\u002522\\u00252C\\u002522extras\\u002522\\u00253A\\u00257B\\u002522fb_app_id\\u002522\\u00253A87741124305\\u00257D\\u00252C\\u002522referer_app_link\\u002522\\u00253A\\u00257B\\u002522url\\u002522\\u00253A\\u002522fb-messenger\\u00253A\\u00255C\\u00252F\\u00255C\\u00252F\\u00255C\\u00252F\\u002522\\u00252C\\u002522app_name\\u002522\\u00253A\\u002522Messenger\\u002522\\u00252C\\u002522package\\u002522\\u00253A\\u002522com.facebook.orca\\u002522\\u00257D\\u00257D&appsite_data=\\u00257B\\u002522android\\u002522\\u00253A\\u00255B\\u00257B\\u002522is_app_link\\u002522\\u00253Atrue\\u00252C\\u002522app_name\\u002522\\u00253A\\u002522YouTube\\u002522\\u00252C\\u002522appsite_url\\u002522\\u00253A\\u002522vnd.youtube\\u00253A\\u00255C\\u00252F\\u00255C\\u00252Fwww.youtube.com\\u00255C\\u00252Fwatch\\u00253Fv\\u00253D7PwQrEbEnrM\\u002526feature\\u00253Dapplinks\\u002522\\u00252C\\u002522fallback_url\\u002522\\u00253A\\u002522https\\u00253A\\u00255C\\u00252F\\u00255C\\u00252Fwww.youtube.com\\u00255C\\u00252Fwatch\\u00253Fv\\u00253D7PwQrEbEnrM\\u002522\\u00252C\\u002522market_uri\\u002522\\u00253A\\u002522market\\u00253A\\u00255C\\u00252F\\u00255C\\u00252Fdetails\\u00253Fid\\u00253Dcom.google.android.youtube\\u002526referrer\\u00253Dutm_source\\u00255Cu00253Dapps.facebook.com\\u00255Cu002526utm_campaign\\u00255Cu00253Dfb4a\\u00255Cu002526utm_content\\u00255Cu00253D\\u00255Cu0025257B\\u00255Cu00252522app\\u00255Cu00252522\\u00255Cu0025253A87741124305\\u00255Cu0025252C\\u00255Cu00252522t\\u00255Cu00252522\\u00255Cu0025253A1501528032\\u00255Cu0025257D\\u002522\\u00252C\\u002522package\\u002522\\u00253A\\u002522com.google.android.youtube\\u002522\\u00257D\\u00255D\\u00257D&has_app_link=1&accessToken=https\\u00253A\\u00252F\\u00252Fwww.youtube.com\\u00252Fwatch\\u00253Fv\\u00253D7PwQrEbEnrM\"}}}"}],"data":{"message_source_data":"{\"message_source\":\"mobile\"}"}}}],"firstDeltaSeqId":691347,"lastIssuedSeqId":691347}
```

**Comment by dequis** [comment]
Yeah, uh, I prefer to repost the relevant part in the ticket with all the personally identifying information removed, but okay. 

The xmaGraphQL:

```
{
  "ZXh0ZW5zaWJsZV9tZXNzYWdlX2F0dGFjaG1lbnQ6MTQ0NTE2Mzk2NTU3MDQ5OQ": {
    "__type__": {
      "name": "ExtensibleMessageAttachment"
    },
    "id": "ZXh0ZW5zaWJsZV9tZXNzYWdlX2F0dGFjaG1lbnQ6MTQ0NTE2Mzk2NTU3MDQ5OQ",
    "is_forwardable": true,
    "story_attachment": {
      "deduplication_key": "ca5f3ea1140edc40d2813149791fb57a",
      "description": {
        "text": "David Gilmour's \"Rattle That Lock\" album released on September 18, 2015 Buy the album at http://smarturl.it/RattleThatLockAmzDLX Watch 'In Any Tongue' here: ..."
      },
      "media": {
        "__type__": {
          "name": "GenericAttachmentMedia"
        },
        "image": {
          "uri": "https://external.fotp3-1.fna.fbcdn.net/safe_image.php?d=AQAN1tyMUSrl4tOj&w=960&h=960&url=https%3A%2F%2Fi.ytimg.com%2Fvi%2F7PwQrEbEnrM%2Fmaxresdefault.jpg&cfs=1&_nc_hash=AQDuWpckE6Kl26DM"
        }
      },
      "properties": [
        {
          "attachment_property_type": "double",
          "key": "width",
          "value": {
            "text": "1920"
          }
        },
        {
          "attachment_property_type": "double",
          "key": "height",
          "value": {
            "text": "1080"
          }
        }
      ],
      "source": {
        "text": "youtube.com"
      },
      "style_list": [
        "video_share_youtube",
        "video_share",
        "share",
        "fallback"
      ],
      "target": {
        "__type__": {
          "name": "ExternalUrl"
        }
      },
      "title": "David Gilmour - The Girl In The Yellow Dress (Official Music Video)",
      "tracking": "{}",
      "url": "fbrpc://facebook/nativethirdparty?app_id=87741124305&app_name=YouTube&class_name=com.google.android.youtube.WatchActivity&fallback_url0=https%3A%2F%2Fm.facebook.com%2Fapps%2Fyt-fb-app&market_uri=market%3A%2F%2Fdetails%3Fid%3Dcom.google.android.youtube%26referrer%3Dutm_source%253Dapps.facebook.com%2526utm_campaign%253Dfb4a%2526utm_content%253D%25257B%252522app%252522%25253A87741124305%25252C%252522t%252522%25253A1501528032%25257D&package_name=com.google.android.youtube&tap_behavior=app_fallback_web&target_url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3D7PwQrEbEnrM&extra_applink_key=al_applink_data&referer_data_key=extras&al_applink_data=%7B%22target_url%22%3A%22https%3A%5C%2F%5C%2Fwww.youtube.com%5C%2Fwatch%3Fv%3D7PwQrEbEnrM%22%2C%22extras%22%3A%7B%22fb_app_id%22%3A87741124305%7D%2C%22referer_app_link%22%3A%7B%22url%22%3A%22fb-messenger%3A%5C%2F%5C%2F%5C%2F%22%2C%22app_name%22%3A%22Messenger%22%2C%22package%22%3A%22com.facebook.orca%22%7D%7D&appsite_data=%7B%22android%22%3A%5B%7B%22is_app_link%22%3Atrue%2C%22app_name%22%3A%22YouTube%22%2C%22appsite_url%22%3A%22vnd.youtube%3A%5C%2F%5C%2Fwww.youtube.com%5C%2Fwatch%3Fv%3D7PwQrEbEnrM%26feature%3Dapplinks%22%2C%22fallback_url%22%3A%22https%3A%5C%2F%5C%2Fwww.youtube.com%5C%2Fwatch%3Fv%3D7PwQrEbEnrM%22%2C%22market_uri%22%3A%22market%3A%5C%2F%5C%2Fdetails%3Fid%3Dcom.google.android.youtube%26referrer%3Dutm_source%5Cu00253Dapps.facebook.com%5Cu002526utm_campaign%5Cu00253Dfb4a%5Cu002526utm_content%5Cu00253D%5Cu0025257B%5Cu00252522app%5Cu00252522%5Cu0025253A87741124305%5Cu0025252C%5Cu00252522t%5Cu00252522%5Cu0025253A1501528032%5Cu0025257D%22%2C%22package%22%3A%22com.google.android.youtube%22%7D%5D%7D&has_app_link=1&accessToken=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3D7PwQrEbEnrM"
    }
  }
}
```

So `fbrpc://facebook/nativethirdparty`, same as https://github.com/bitlbee/bitlbee-facebook/issues/97

I think I have a fix for this somewhere in the queue. Weird that there were no tickets for it in this tracker already.

**Comment by kaolincash** [comment]
I actually just came here to make this ticket, myself, so perhaps that's a little less weird.
Glad to know about a possible fix, I'll keep an eye out.

Sidenote: I had trouble with facebook images, too, so thanks for the fix on that one. Eion didn't get back to me before I found the ticket, and I was amused to see you tagged him on it yourself.

**Comment by db-bogdan** [comment]
@dequis do you have an estimated date for any fix for this at any chance? Just wondering

**Comment by dequis** [comment]
This is fixed in 0.9.5

https://github.com/dequis/purple-facebook/releases/tag/v0.9.5-9ff9acf9fa14
